src/Umbraco.Core/Models/Blocks/BlockGridItem.cs

```diff
@@ -0,0 +1,164 @@
+using System;
+using System.Runtime.Serialization;
+using Umbraco.Cms.Core.Models.PublishedContent;
+
+namespace Umbraco.Cms.Core.Models.Blocks
+{
+    /// <summary>
+    /// Represents a layout item for the Block Grid editor.
+    /// </summary>
+    /// <seealso cref="Umbraco.Core.Models.Blocks.IBlockReference{Umbraco.Core.Models.PublishedContent.IPublishedElement}" />
+
+    // TODO: Change: Niels: this is also called block, be aware!
+    [DataContract(Name = "block", Namespace = "")]
+    public class BlockGridItem : IBlockReference<IPublishedElement>
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="BlockGridItem" /> class.
+        /// </summary>
+        /// <param name="contentUdi">The content UDI.</param>
+        /// <param name="content">The content.</param>
+        /// <param name="settingsUdi">The settings UDI.</param>
+        /// <param name="settings">The settings.</param>
+        /// <param name="rowSpan">The number of rows to span</param>
+        /// <param name="columnSpan">The number of columns to span</param>
+        /// <exception cref="System.ArgumentNullException">contentUdi
+        /// or
+        /// content</exception>
+        public BlockGridItem(Udi contentUdi, IPublishedElement content, Udi settingsUdi, IPublishedElement settings, int rowSpan, int columnSpan)
+        {
+            ContentUdi = contentUdi ?? throw new ArgumentNullException(nameof(contentUdi));
+            Content = content ?? throw new ArgumentNullException(nameof(content));
+            SettingsUdi = settingsUdi;
+            Settings = settings;
+            RowSpan = rowSpan;
+            ColumnSpan = columnSpan;
+        }
+
+        /// <summary>
+        /// Gets the content UDI.
+        /// </summary>
+        /// <value>
+        /// The content UDI.
+        /// </value>
+        [DataMember(Name = "contentUdi")]
+        public Udi ContentUdi { get; }
+
+        /// <summary>
+        /// Gets the content.
+        /// </summary>
+        /// <value>
+        /// The content.
+        /// </value>
+        [DataMember(Name = "content")]
+        public IPublishedElement Content { get; }
+
+        /// <summary>
+        /// Gets the settings UDI.
+        /// </summary>
+        /// <value>
+        /// The settings UDI.
+        /// </value>
+        [DataMember(Name = "settingsUdi")]
+        public Udi SettingsUdi { get; }
+
+        /// <summary>
+        /// Gets the settings.
+        /// </summary>
+        /// <value>
+        /// The settings.
+        /// </value>
+        [DataMember(Name = "settings")]
+        public IPublishedElement Settings { get; }
```

Comment:
@nielslyngsoe should SettingsUdi and Settings be nullable?

Reply:
I would say yes, but lets see what @kjac says about that :-)

Reply:
Ideally yes - it would make a lot of sense, since the settings data might not be there. However, `Settings` are defined in the interface `IBlockReference<TSettings>` as not nullable, and that interface is shared between `BlockGridItem` and `BlockListItem`. In other words we'd introduce a breaking change for block list on `BlockListItem`, which is not entirely ideal at this point.